src/error.rs

```diff
@@ -59,6 +59,9 @@ pub enum Error {
     /// UTF coding error.
     #[error("UTF-8 encoding error")]
     Utf8,
+    /// Attack attempt detected.
+    #[error("Attack attempt detected")]
+    AttackAttempt,
```

Comment:
I think we could use a bit more descriptive name, perhaps something like `HeadersTooLarge(usize)` or similar, otherwise, it may be a bit hard to tell what went wrong in a rare case when the user bug (an invalid client that sends large headers for some obscure reasons).

Reply:
This is intended to use for any suspicious activity, not only for headers that are too large. For example, sending a medium-size header in 1-byte packets is bad for the current `httparse` and results in the same error. We could deprecate and rename this later after improving HTTP parsing in general.